orttraining/orttraining/test/distributed/pipeline_partition_test.cc

```diff
@@ -0,0 +1,227 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "orttraining/core/graph/pipeline_transformer.h"
+#include "core/graph/model.h"
+
+#include "core/common/path_utils.h"
+#include "test/test_environment.h"
+
+#include "gtest/gtest.h"
+
+#include "partition_utils.hpp"
+
+namespace onnxruntime {
+namespace test {
+
+using namespace onnxruntime;
+using namespace onnxruntime::common;
+using namespace onnxruntime::training;
+
+
+using CutList = std::vector<TrainingSession::TrainingConfiguration::CutInfo>;
+
+TEST(PipelinePartition, DropoutGraph2stages) {
+  int num_stages = 2;
+  int pipeline_stage_id = 1;
+  std::map<std::string, int> input_map = {
+    {"A", 0}, {"B", 0}, {"C", 0}, {"D", 0}, {"E", 0}, {"F", 0},
+    {"O1", 1}, {"O2", 1}, {"O3", 1}, {"O4", 1}, {"O5", 1}
+  };
+
+  const auto& log_manager = DefaultLoggingManager();
+  const auto& default_logger = log_manager.DefaultLogger();
+  const auto model_path = ORT_TSTR("testdata/transform/dropout.onnx");
+
+  std::shared_ptr<Model> pModel;
+  auto status = Model::Load(model_path, pModel, nullptr, default_logger);
+  EXPECT_TRUE(status.IsOK()) << "Failed to load model. Error: "
+                             << status.ErrorMessage();
+  auto& graph = pModel->MainGraph();
+
+  std::map<Node*, int> op_to_stage = {};
+  status = GetDeviceAssignmentMap(graph, input_map, op_to_stage, num_stages);
+
+  EXPECT_TRUE(status.IsOK()) << "Failed to get stage map. Error: "
+                             << status.ErrorMessage();
+  status = ApplyPipelinePartitionToMainGraph(graph, op_to_stage,
+                                             pipeline_stage_id, num_stages);
+  EXPECT_TRUE(status.IsOK()) << "Failed to apply partition. Error: "
+                             << status.ErrorMessage();
+
+  // Stage 1 should have 6 nodes: 6 Identities + 1 receive.
+  EXPECT_EQ(graph.NumberOfNodes(), 6);
+}
+
+void LoadAndPartitionWithCuts(const PathString& filename,
+                              int num_stages,
+                              int pipeline_stage_id,
+                              CutList& cuts,
+                              bool use_stage_map,
+                              std::shared_ptr<Model>& pModel) {
+  const auto& log_manager = DefaultLoggingManager();
+  const auto& default_logger = log_manager.DefaultLogger();
+  const auto model_path = ORT_TSTR(filename);
```

Comment:
Remember to remove the ORT_TSTR macro here. The macro makes sense only for string literals.

Reply:
yes, thanks. I forgot to remove this one before pushing.